lib/terminal.robot

```diff
@@ -63,7 +63,13 @@ Read From Terminal Until
     ...    connection to the DUT (Telnet or SSH).
     [Arguments]    ${expected}
     IF    '${DUT_CONNECTION_METHOD}' == 'Telnet'
-        ${output}=    Telnet.Read Until    ${expected}
+        TRY
+            ${output}=    Telnet.Read Until    ${expected}
+        EXCEPT    EOFError: telnet connection closed
+            ${output}=    Fail
+            ...    - telnet (or minicom on RTE) connection might be opened- either by you or someone else\n- Verify and close your connections before starting tests\n- stop minic
+            Om Processes On RTE, Restart Ser2net Service (systemctl Restart Ser2net)'
```

Comment:
```suggestion
            ${err_msg}=    Catenate    SEPARATOR=\n
            ...    - telnet (or minicom on RTE) connection might be opened - either by you or someone else
            ...    - Verify and close your connections before starting tests
            ...    - stop minicom processes on RTE, restart ser2net service (systemctl restart ser2net)
            Fail    ${err_msg}
```

1. You don't need to save ${output} of Fail
2. You need to check pre-commit changes before adding them (line 71)
3. I suggest using Catenate to create long multiline string, it will be much more readable.